yt_dlp/extractor/crunchyroll.py

```diff
@@ -86,9 +91,13 @@ def _update_auth(self):
 
         grant_type = 'etp_rt_cookie' if self.is_logged_in else 'client_id'
         try:
+            auth_headers = {'Authorization': CrunchyrollBaseIE._BASIC_AUTH}
+            if not self.is_logged_in:
+                auth_headers['ETP-Anonymous-ID'] = uuid.uuid4()
```

Comment:
This should probably be outside the `try`. Perhaps
```suggestion
        auth_headers = {'Authorization': CrunchyrollBaseIE._BASIC_AUTH}
        if self.is_logged_in:
            grant_type = 'etp_rt_cookie'
        else:
            grant_type = 'client_id'
            auth_headers['ETP-Anonymous-ID'] = uuid.uuid4()
        try:
```

`uuid` is required now?

Reply:
It's not required, the value can be anything non-empty but the device id is (at least on the browser) a uuid so I just mimic that

Reply:
Should this be read as "The ETP-Anonymous-ID header is not required at all" or as "The value is not required to be an uuid4"? If the former maybe we shouldnt be setting it at all?